test/list_test.dart

```diff
@@ -724,4 +725,31 @@ Future<void> main([List<String>? args]) async {
     expect(team.players.length, 10);
     expect(realm.all<Person>().length, 100);
   });
+
+  test('ManagedRealmList.indexOf', () {
+    final config = Configuration.local([Team.schema, Person.schema]);
+    final realm = getRealm(config);
+
+    final team = Team('sad team', players: [for (int i = 0; i < 100; ++i) Person('$i')]);
+    realm.write(() => realm.add(team));
+    final players = team.players;
+
+    // expect(players, isA<ManagedRealmList<Person>>());
+    expect(players.indexOf(players.first, -1), 0); // okay to start from negative index
+    expect(players.indexOf(players.first, 1), -1); // start respected
+    expect(players.indexOf(players.first, 101), -1); // okay to start from non-existent index
+
+    var index = 0;
+    final r = Random(42); // deterministic
+    for (final p in players) {
+      expect(players.indexOf(p, r.nextInt(index + 1) - 1), index++);
```

Comment:
Can we add a test that uses `indexOf` for an unmanaged object (i.e. `players.indexOf(Person('a')`).